rest/blip_client_test.go

```diff
@@ -716,3 +716,34 @@ func (btr *BlipTesterReplicator) storeMessage(msg *blip.Message) {
 	defer btr.messagesLock.Unlock()
 	btr.messages[msg.SerialNumber()] = msg
 }
+
+// Returns the BLIP message by profile.
+func (btc *BlipTesterClient) GetMessage(profile string) (msg *blip.Message, found bool) {
```

Comment:
This is only going to work when there is a single message per profile in a test. When you get a test that sends 2 `rev` messages, how do you wait for the first, and then the second?

Looking at _why_ the test is waiting, it needs to first wait for the given document revision to be stored by the client, and then fetch the associated BLIP message responsible for creating that revision to do some assertions on it.

Maybe we can store a reference to the blip.Message responsible for creating a document on the `BlipTesterReplicator` somewhere, as a more reliable method than the `messages` map.

Reply:
This change has been modified quite a bit; Both the message reference and request body bytes are wrapped inside `BodyMessagePair` struct which can be stored inside the client's local store of documents: `map(docId -> map(revId -> BodyMessagePair))`.